src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.cs

```diff
@@ -1,69 +1,268 @@
+                        return;
+                    }
+
+                    var matchedMethods = new List<IMethodSymbol>();
+                    var unmatchedMethods = new HashSet<string>() { expectedGroup.AlternateMethod1 };
+                    if (expectedGroup.AlternateMethod2 != null)
+                    {
+                        unmatchedMethods.Add(expectedGroup.AlternateMethod2);
+                    }
+
+                    foreach (var candidateMethod in typeSymbol.GetMembers().OfType<IMethodSymbol>())
+                    {
+                        if (candidateMethod.Name == expectedGroup.AlternateMethod1 || candidateMethod.Name == expectedGroup.AlternateMethod2)
+                        {
+                            // found an appropriately-named method
+                            matchedMethods.Add(candidateMethod);
+                            unmatchedMethods.Remove(candidateMethod.Name);
+                        }
+                    }
+
+                    // only one public method match is required
+                    if (matchedMethods.Any(m => m.IsPublic()))
+                    {
+                        // at least one public alternate method was found, do nothing
+                    }
+                    else
+                    {
+                        // either we found at least one method that should be public or we didn't find anything
+                        var notPublicMethod = matchedMethods.FirstOrDefault(m => !m.IsPublic());
+                        if (notPublicMethod != null)
+                        {
+                            // report error for improper visibility directly on the method itself
+                            symbolContext.ReportDiagnostic(CreateDiagnostic(VisibilityRule, GetSymbolLocation(notPublicMethod), FixVisibilityText, notPublicMethod.Name, operatorName));
+                        }
+                        else
+                        {
+                            // report error for missing methods on the operator overload
+                            if (expectedGroup.AlternateMethod2 == null)
+                            {
+                                // only one alternate expected
+                                symbolContext.ReportDiagnostic(CreateDiagnostic(DefaultRule, GetSymbolLocation(methodSymbol), AddAlternateText, expectedGroup.AlternateMethod1, operatorName));
+                            }
+                            else
+                            {
+                                // one of two alternates expected
+                                symbolContext.ReportDiagnostic(CreateDiagnostic(MultipleRule, GetSymbolLocation(methodSymbol), AddAlternateText, expectedGroup.AlternateMethod1, expectedGroup.AlternateMethod2, operatorName));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static Location GetSymbolLocation(ISymbol methodSymbol)
```

Comment:
Nitpick: It's not always a method symbol, sometimes a property, so name s.b. just `symbol`.